### PR TITLE
Speed up actor creation task submission by generating IDs with uuid.

### LINKED ISSUE #1698: Speed up actor ID (and other random ID generation).
This function is really slow (on the order of 5-10ms). It is used in a bunch of places and should be replaced with something faster (e.g., a deterministic hash). This is an issue e.g., for actor creation.

https://github.com/ray-project/ray/blob/cae108d019a51bd7bd35df7f93b68ff07a84d9cd/python/ray/utils.py#L84-L107

### PR DESCRIPTION
**Pull request by robertnishihara**
This speeds up actor creation (just the submission part) by 10x. See the explanation in #1698.

```python
import ray
ray.init()

@ray.remote
class Foo(object):
    pass

Foo.remote()  # The first time is a bit slower since the actor is exported here.
%time Foo.remote()  # This changes from 6ms to 0.6ms
```

Fixes #1698.

### CODE REVIEW

**Comment by stephanie-wang** [comment]
Hmm is it possible to reuse the task ID of the actor creation task as the actor ID?

**Comment by robertnishihara** [comment]
@stephanie-wang that would work. Are you suggesting this so we can potentially remove some fields from the TaskSpec?

**Comment by robertnishihara** [comment]
I switched back to not generating actor IDs deterministically for now.. Also adopted @richardliaw's suggestion of using `uuid`. I *think* that won't mess up any random number generator states, but I'm not sure..

**Comment by richardliaw** [comment]
```
In [17]: import random

In [18]: random.seed(1)

In [19]: random.random()
Out[19]: 0.13436424411240122

In [20]: uuid.uuid4().bytes
Out[20]: b'E\xb8\xdd`\xaf\xd9N\xa2\xaah\x846\x1e"n\xe2'

In [21]: random.random()
Out[21]: 0.8474337369372327

In [22]:

In [22]: random.seed(1)

In [23]:  random.random()
Out[23]: 0.13436424411240122

In [24]:  random.random()
Out[24]: 0.8474337369372327
```

**Comment by stephanie-wang** [comment]
@robertnishihara, I'm actually not sure about whether we can remove extra fields. I was mostly just thinking that it makes sense to just reuse the task ID since it's already deterministically computed.

My preference would be if we used the actor creation task ID as the actor ID, and also replaced the `ray.utils.random_string` implementation with `uuid`.

**Comment by robertnishihara** [comment]
I think we should go ahead and merge this as is since there's no change in behavior and it speeds things up. I'll create a separate issue for making ID generation deterministic (which I agree is critical).

File: python/ray/actor.py [hunk 1]
```diff
@@ -7,21 +7,29 @@
...
 import ray.signature as signature
 import ray.worker
-from ray.utils import (FunctionProperties, random_string, is_cython,
-                       push_error_to_driver)
+from ray.utils import FunctionProperties, is_cython, push_error_to_driver
 
 
 def random_actor_id():
-    return ray.local_scheduler.ObjectID(random_string())
+    actor_id_hash = hashlib.sha1()
```
Comment (pcmoritz) [inline python/ray/actor.py]: Let's factor these out into a common method

**Comment by robertnishihara** [comment]
See #1751.

**Review by pcmoritz — APPROVED** [review]

**Pull request merged**